### PR TITLE
Link Component Refactor

### PR DESCRIPTION
**Pull request by georgeweiler**
- Removes `InternalLink` and `ExternalLink` components
- Adds a `Link` component with correct styling and `isExternalProp`
- Refactors all uses of `ExternalLink` to use new `Link` component. Replaces `text` prop with children
- Adds `ButtonLink` component with `isExternal` prop, and updates usages
- Removes unnecessary of `as={RouterLink}`

Fixes: https://github.com/threshold-network/token-dashboard/issues/228
Fixes: https://github.com/threshold-network/token-dashboard/issues/230

### CODE REVIEW

File: src/components/Link/index.tsx [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+
+type LinkHrefProps =
+  | { isExternal?: false; href?: never }
+  | { isExternal: true; href: string }
+
+type LinkToProps =
+  | { isExternal?: false; to: string }
+  | { isExternal: true; to?: never }
+
+type Props = CommonProps & LinkHrefProps & LinkToProps
```
Comment (michalsmiarowski) [inline src/components/Link/index.tsx:24]: Can we export these types here and use them in `ButtonLink` too?

If you agree with exporting them then let's also change the names:
`CommonProps` -> `CommonLinkProps`
`Props` -> `LinkProps`

File: src/components/StakingTimeline/index.tsx [hunk 2]
```diff
@@ -81,7 +78,7 @@ export const LegacyStakesDepositSteps: FC = () => {
             Enter the Provider, Beneficiary, and Authorizer addresses. These
             will be automatically set to your wallet address. If you want to use
             a Staking Provider, here is{" "}
-            <InternalLink to={STAKING_PROVIDER_URL}>a list</InternalLink>
+            <Link to={STAKING_PROVIDER_URL}>a list</Link>
```
Comment (michalsmiarowski) [inline src/components/StakingTimeline/index.tsx:81]: Non-blocking:

Do you think we should add dot here at the end?

```suggestion
            <Link to={STAKING_PROVIDER_URL}>a list</Link>.
```

File: src/components/StakingTimeline/index.tsx [hunk 3]
```diff
@@ -107,10 +104,8 @@ export const LegacyStakesDepositSteps: FC = () => {
         >
           You will need to run a node for applications that you have authorized
           to earn rewards. If you don’t have one, learn how to do it{" "}
-          <InternalLink to={APPLICATION_DOCS_URL}>here</InternalLink> or contact{" "}
-          <InternalLink to={STAKING_PROVIDER_URL}>
-            a Staking Provider
-          </InternalLink>
+          <Link to={APPLICATION_DOCS_URL}>here</Link> or contact{" "}
+          <Link to={STAKING_PROVIDER_URL}>a Staking Provider</Link>
```
Comment (michalsmiarowski) [inline src/components/StakingTimeline/index.tsx:108]: Non-blocking:

Sam as above:

```suggestion
          <Link to={STAKING_PROVIDER_URL}>a Staking Provider</Link>.
```

File: src/components/StakingTimeline/index.tsx [hunk 4]
```diff
@@ -240,11 +224,8 @@ const StakingTimeline: FC<{ statuses?: FlowStepStatus[] } & StackProps> = ({
         >
           You will need to run a node for applications that you have authorized
           to earn rewards. If you don’t have one, learn how to do it{" "}
-          <InternalLink to={APPLICATION_DOCS_URL}>here</InternalLink> or contact
-          a{" "}
-          <InternalLink to={STAKING_PROVIDER_URL}>
-            Staking Provider.
-          </InternalLink>
+          <Link to={APPLICATION_DOCS_URL}>here</Link> or contact a{" "}
+          <Link to={STAKING_PROVIDER_URL}>Staking Provider.</Link>
```
Comment (michalsmiarowski) [inline src/components/StakingTimeline/index.tsx:228]: Non-blocking:

Let's move the dot outside of `Link` component:

```suggestion
          <Link to={STAKING_PROVIDER_URL}>Staking Provider</Link>.
```

**Comment by michalsmiarowski** [comment]
I thinkt i would give an approve. Let's wait for @r-czajkowski 's review before we merge it

File: src/pages/Overview/Network/StakingOverview.tsx [hunk 5]
```diff
@@ -23,9 +24,9 @@ const StakingOverview: FC = () => {
           isLarge
         />
       </InfoBox>
-      <Button size="lg" isFullWidth mt={4} as={RouterLink} to="/staking">
+      <ButtonLink size="lg" isFullWidth mt={4} to="/something">
```
[the remark this answers is not in the document]
Comment (r-czajkowski) [reply src/pages/Overview/Network/StakingOverview.tsx]: ade0d6d

File: src/components/Link/index.tsx [hunk 6]
```diff
@@ -0,0 +1,58 @@
...
+  icon?: ReactElement
+}
+
+type LinkHrefProps =
+  | { isExternal?: false; href?: never }
+  | { isExternal: true; href: string }
+
+type LinkToProps =
+  | { isExternal?: false; to: string }
+  | { isExternal: true; to?: never }
```
[the remark this answers is not in the document]
Comment (r-czajkowski) [reply src/components/Link/index.tsx]: 6091620

File: src/components/ButtonLink/index.tsx [hunk 7]
```diff
@@ -0,0 +1,37 @@
...
+            <Icon boxSize="12px" ml="1" as={FiArrowUpRight} />
+          ) : undefined
+        }
+        {...props}
+      >
+        {children}
+      </Button>
+    )
+  }
+)
```
[the remark this answers is not in the document]
Comment (r-czajkowski) [reply src/components/ButtonLink/index.tsx]: 8d3dcb8

**Review by r-czajkowski — APPROVED** [review]

**Pull request merged**